shared/macros-oval.jinja

```diff
@@ -0,0 +1,236 @@
+{{#
+  High level macro which checks if a particular combination of parameter and value in a configuration file is set.
+  This function can take nine parameters:
+    - path (String): Path to the configuration file to be checked.
+    - prefix_regex (String): Regular expression to be used in the beginning of the OVAL text file content check.
+    - parameter (String): The parameter to be checked in the configuration file.
+    - value (String): The value to be checked. This can also be a regular expression (e.g: value1|value2 can match both values).
+    - separator_regex (String): Regular expression to be used as the separator of parameter and value in a configuration file. If spaces are allowed, this should be included in the regular expression.
+    - default_pass (boolean): Implies that the value for the parameter is the default value. If set, the check will pass even if the parameter is not set in the configuration file.
+    - application (String): The application which the configuration file is being checked. Can be any value and does not affect the OVAL check.
+    - multi_value (boolean): If set, it means that the parameter can accept multiple values and the expected value must be present in the current list of values.
+    - missing_config_file_fail (boolean): If set, the check will fail if the configuration is not existent in the system.
+    - supported_platforms (String List): List of supported platforms the OVAL will applicable to. (e.g.: ['rhel7', 'rhel8', 'fedora'])
+#}}
+{{%- macro oval_check_config_file(path='', prefix_regex='^\s*', parameter='', separator_regex='(?-i)[\s]+', value='', default_pass='', application='', multi_value='', missing_config_file_fail=false, supported_platforms=['all']) -%}}
+<def-group>
+  <definition class="compliance" id="{{{ rule_id }}}" version="1">
+    <metadata>
+      <title>{{{ rule_title }}}</title>
+      <affected family="unix">
+{{#-
+  Use macro oval_affected(products) when it is available
+#}}
+        <platform>multi_platform_rhel</platform>
+        <platform>multi_platform_fedora</platform>
+        <platform>multi_platform_rhv</platform>
+        <platform>multi_platform_ol</platform>
+      </affected>
+      <description>{{{ rule_description }}}</description>
```

Comment:
After checking the built DataStream, I'm not fond of having rule description here.
It can be too long, and doesn't add much value.

As the OVAL description, shomething like what was in the definition for `grub2_nousb_argument` is more useful:
`Look for 'nousb' argument in the kernel line in /etc/default/grub`

Reply:
I changed to macro: `Look for parameter '{{{ parameter }}}' with value '{{{ value }}}' in '{{{ path }}}'`